catroid/src/main/java/org/catrobat/catroid/ui/fragment/CategoryBricksFactory.kt

```diff
@@ -0,0 +1,767 @@
+        return arduinoBrickList
+    }
+
+    private fun setupRaspiCategoryList(): List<Brick> {
+        val defaultScript = RaspiInterruptScript("3", "pressed")
+        val raspiBrickList: MutableList<Brick> = ArrayList()
+        raspiBrickList.add(WhenRaspiPinChangedBrick(defaultScript))
+        raspiBrickList.add(RaspiIfLogicBeginBrick(Formula(BrickValues.RASPI_DIGITAL_INITIAL_PIN_NUMBER)))
+        raspiBrickList.add(RaspiSendDigitalValueBrick(BrickValues.RASPI_DIGITAL_INITIAL_PIN_NUMBER, BrickValues.RASPI_DIGITAL_INITIAL_PIN_VALUE))
+        raspiBrickList.add(RaspiPwmBrick(BrickValues.RASPI_DIGITAL_INITIAL_PIN_NUMBER, BrickValues.RASPI_PWM_INITIAL_FREQUENCY, BrickValues.RASPI_PWM_INITIAL_PERCENTAGE))
+        return raspiBrickList
+    }
+
+    private fun setupEmbroideryCategoryList(context: Context): List<Brick> {
+        val embroideryBrickList: MutableList<Brick> = ArrayList()
+        embroideryBrickList.add(StitchBrick())
+        embroideryBrickList.add(SetThreadColorBrick(Formula(BrickValues.THREAD_COLOR)))
+        embroideryBrickList.add(RunningStitchBrick(Formula(BrickValues.STITCH_LENGTH)))
+        embroideryBrickList.add(ZigZagStitchBrick(Formula(BrickValues.ZIGZAG_STITCH_LENGTH), Formula(BrickValues.ZIGZAG_STITCH_WIDTH)))
+        embroideryBrickList.add(TripleStitchBrick(Formula(BrickValues.STITCH_LENGTH)))
+        embroideryBrickList.add(SewUpBrick())
+        embroideryBrickList.add(StopRunningStitchBrick())
+        embroideryBrickList.add(WriteEmbroideryToFileBrick(context.getString(R.string.brick_default_embroidery_file)))
+        return embroideryBrickList
+    }
+
+    private fun setupAssertionsCategoryList(context: Context): List<Brick> {
+        val assertionsBrickList: MutableList<Brick> = ArrayList()
+        assertionsBrickList.add(AssertEqualsBrick())
+        assertionsBrickList.add(AssertUserListsBrick())
+        assertionsBrickList.add(ParameterizedBrick())
+        assertionsBrickList.add(WaitTillIdleBrick())
+        assertionsBrickList.add(TapAtBrick(BrickValues.TOUCH_X_START, BrickValues.TOUCH_Y_START))
+        assertionsBrickList.add(TapForBrick(BrickValues.TOUCH_X_START, BrickValues.TOUCH_Y_START, BrickValues.TOUCH_DURATION))
+        assertionsBrickList.add(TouchAndSlideBrick(BrickValues.TOUCH_X_START, BrickValues.TOUCH_Y_START, BrickValues.TOUCH_X_GOAL, BrickValues.TOUCH_Y_GOAL, BrickValues.TOUCH_DURATION))
+        assertionsBrickList.add(FinishStageBrick())
+        assertionsBrickList.add(StoreCSVIntoUserListBrick(BrickValues.STORE_CSV_INTO_USERLIST_COLUMN, context.getString(R.string.brick_store_csv_into_userlist_data)))
+        assertionsBrickList.add(WebRequestBrick(context.getString(R.string.brick_web_request_default_value)))
+        return assertionsBrickList
+    }
+
+    fun searchList(searchBrick: Brick, list: List<Brick>): Boolean = list.any { it == searchBrick.javaClass }
+
+    fun getBrickCategory(brick: Brick, isBackgroundSprite: Boolean, context: Context): String? {
```

Comment:
why nullable string as return type?
```suggestion
    fun getBrickCategory(brick: Brick, isBackgroundSprite: Boolean, context: Context): String {
```